localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_task/state_task.py

```diff
@@ -69,13 +69,13 @@ def _eval_parameters(self, env: Environment) -> dict:
 
         return parameters
 
-    def _eval_credentials(self, env: Environment) -> ComputedCredentials:
+    def _eval_state_credentials(self, env: Environment) -> StateCredentials:
         if not self.credentials:
-            task_credentials = dict()
+            state_credentials = StateCredentials(role_arn=env.aws_execution_details.role_arn)
         else:
             self.credentials.eval(env=env)
```

Comment:
Could this fail? if so, what would/should be the behavior here then? :thinking: 

Also curious if (not here) a failed assume role would also lead to falling back to the role ARN potentially? I'd assume not, since that might be a security issue.

Reply:
The computation of the user-defined credentials block can fail, we have some test cases to validate our handling of such failures `aws.services.stepfunctions.v2.credentials.test_credentials_base.TestCredentialsBase.test_invalid_credentials_field`
In the tandem PR, which adds tests for SFN IAM enforcement, I did not see this behaviour, instead the failure is simply passed upstream